client/src/components/OfflineWarning/offline-warning.tsx

```diff
@@ -1,16 +1,20 @@
 import React from 'react';
-import PropTypes from 'prop-types';
 import { useTranslation } from 'react-i18next';
 
 import './offline-warning.css';
 
 const delayInMilliSeconds = 5000;
-let id;
-const propTypes = {
-  isOnline: PropTypes.bool.isRequired,
-  isSignedIn: PropTypes.bool.isRequired
-};
-function OfflineWarning({ isOnline, isSignedIn }) {
+let id: NodeJS.Timeout;
```

Comment:
```suggestion
let id: number;
```

Reply:
Ah, this was just what my IDE recommended as a return type.

I get a TS error when using `number`, let me know what you think of my change to `ReturnType<typeof setTimeout>` instead!

Reply:
I think this is fine. We *could* use window.setTimeout, but then we'd need to check if window existed before using it. This approach avoids that mess, which is a plus.